capstone-app/src/app/modules/photos/components/image-container/image-container.component.ts

```diff
@@ -1,39 +1,66 @@
 import { Photo } from 'src/app/modules/photos/photo-template';
 import { PhotoFetcher } from 'src/app/modules/photos/services/photo-fetcher.service';
-import { Component, OnInit, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import {
+  Component,
+  Input,
+  OnChanges,
+  Output,
+  EventEmitter,
+  OnInit,
+} from '@angular/core';
 
 @Component({
   selector: 'app-image-container',
   templateUrl: './image-container.component.html',
   styleUrls: ['./image-container.component.scss'],
 })
-export class ImageContainerComponent implements OnInit, OnChanges  {
+export class ImageContainerComponent implements OnChanges, OnInit {
   displayPhotos: Photo[] = [];
   originalPhotos: Photo[] = [];
-  query: string;
   errorMessage: string;
+  query = '';
+  previousCity = '';
+  previousFilter = '';
 
   /**
-   * @Input limit - Receives count filter from parent on how many
-   * photos to display
-   * @Output limitChange - When an image cannot be displayed,
-   * tell the parent to update image count
+   * @Input city: the city user searched for.
+   * @Input filter: the filter user clicked on.
+   * @Input limit: the photo number to display.
+   * @Output limitChange: if a photo cannot display, update limit.
    */
-  @Input() limit: number;
-  @Output() limitChange = new EventEmitter<number>();
+  @Input() city: string;
```

Comment:
I'd recommend specifying if the city and filter inputs are required or optional, by using either '!' or '?'. This is standard practice at google, but not necessarily in the outside world.